Add name history command - Adds #900

lemonsked opened the pull request:
Adds #900

lemonsked committed:
Create FHistory.java

lemonsked committed:
Create History.java

lemonsked committed:
Create UUIDFetcher.java

lemonsked committed:
Create Command_namehistory

lemonsked committed:
Rename Command_namehistory to Command_namehistory.java

lemonsked commented:
You have to use the latest username of the player- Can't use an earlier username.

lemonsked committed:
Formatting

Telesphoreo commented:
Tested on my server and it works

Wild1145 commented:
The related issue is for the 5.2 release, I have placed this PR on hold and it can be re-reviewed when we have completed work on the 5.1 release.

lemonsked commented:
@JeromSar

lemonsked commented:
Oops I accidently closed it

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+
+                    if (history == null)
+                    {
+                        sender.sendMessage(ChatColor.RED + "Player not found!");
```
As the Bukkit API is not thread safe, you cannot call this method asynchronously. You have to resynchronise with the server first.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+
+            for (int i = 1; i < oldNames.length; i++)
+            {
+                DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```
Seems like something that should be a field, probably static and final.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+    }
+}
+
+class FName implements Comparable<FName>
```
Does this compile? I am sure you cannot have two classes in one file. Either way. Declare public/private.

lemonsked replied:
You can have more then two classes in one file. MainConfig does this for the ConfigDefaults class.

JeromSar replied:
Prett sure those are nested static classes, and not seperate classes.

lemonsked replied:
Well, it compiles, and I tested it on my server, and it works.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/UUIDFetcher.java:
```diff
@@ -0,0 +1,63 @@
+import java.net.URL;
+import java.util.UUID;
+
+class FetchedUuid
```
Once again, is this possible? Declare private/public.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/UUIDFetcher.java:
```diff
@@ -0,0 +1,63 @@
+class FetchedUuid
+{
+
+    String id;
```
This field should probably be final, and should be private or public.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+                    }
+                    catch (Exception ex)
+                    {
+                        FLog.severe(ex);
```
When catching this, should probably send a message to the CommandSender as well, to notify them something went wrong.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+        {
+            sender.sendMessage(ChatColor.GREEN + oldNames[0].getName() + ChatColor.GOLD + " has never changed their name.");
+        }
+        else
```
return, instead of else block?

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+
+class FName implements Comparable<FName>
+{
+    public String name;
```
This field has a getter, so this should probably be private.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+class FName implements Comparable<FName>
+{
+    public String name;
+    public long changedToAt;
```
This field has no getter, but should probably have one. This means that this should probably be a private field.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,98 @@
+                        conn.disconnect();
+                        Arrays.sort(oldNames);
+                        history = new FHistory(uuid, oldNames);
+                        printHistory(sender, oldNames);
```
As the Bukkit API is not thread safe, you cannot call this method asynchronously. You have to resynchronise with the server first.

JeromSar commented:
Looks good, but needs some changes, see above.

lemonsked committed:
Update History.java

lemonsked committed:
Update History.java

lemonsked committed:
Update History.java

lemonsked committed:
Update History.java

lemonsked committed:
Update UUIDFetcher.java

lemonsked commented:
Fixed @JeromSar

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,102 @@
+    
+    public static void reportHistory(final CommandSender sender, final String username)
+    {
+        Player player = Bukkit.getPlayer(sender.getName());
```
This does not work when the CommandSender is a console. Instead, perhaps change `FSync.playerMsg()` to take a CommandSender as an argument?

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,102 @@
+            {
+                UUID uuid;
+                FHistory history = null;
+                uuid = UUIDFetcher.fetch(username);
```
Might as well declare uuid here, as the declaration above has no other purpose.

JeromSar reviewed (changes requested):
Please change the PR to reflect the comments above and below.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -15,11 +15,16 @@
 
 public class History
 {
+    public static DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```
Perhaps this should be final?

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/UUIDFetcher.java:
```diff
@@ -9,10 +9,10 @@
 {
 
-    String id;
+    public final String id;
```
Perhaps this needs a getter.

lemonsked committed:
Update FSync.java

lemonsked committed:
Update History.java

lemonsked committed:
Update UUIDFetcher.java

lemonsked commented:
Done changes @JeromSar

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,99 @@
+{
+    public static final DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    
+    public static void reportHistory(final CommandSender sender, final String username
```
You are missing another parentheses.

lemonsked committed:
Update History.java

lemonsked committed:
Update History.java

lemonsked commented:
@JeromSar ?

lemonsked committed:
Update UUIDFetcher.java

lemonsked committed:
Update FSync.java

lemonsked committed:
Update History.java

lemonsked committed:
Delete FHistory.java

lemonsked committed:
Don't need import -> Same package.

Telesphoreo commented:
I was able to test lemons latest code pushed and it all works

JeromSar commented:
@OxLemonxO Testing this ASAP. merging when tested.

lemonsked commented:
Bump?

JeromSar commented:
Debump.

Commit:
:boom::camel: Added .gitattributes & .gitignore files

Commit:
TFM massive patches

TFM patches

Wild1145 commented:
Currently not building https://travis-ci.org/Wild1145/TotalFreedomMod/builds/230516942?utm_source=github_status&utm_medium=notification

Commit:
Csnnot be private

Commit:
Cannot be private

Wild1145 commented:
Automated builds are now working again.

Commit:
Updated

Commit:
Updated again

StevenLawson committed:
Stuff that marco changed...

StevenLawson committed:
Make httpd players module only show active admins.

StevenLawson committed:
Fix a concurrent modification exception & logviewer removals.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/UUIDFetcher.java:
```diff
@@ -0,0 +1,67 @@
+import java.net.URL;
+import java.util.UUID;
+
+class FetchedUuid
```
Declare public please.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,101 @@
+    }
+}
+
+class FName implements Comparable<FName>
```
Please convert this to a nested static class.

Commit:
Convert to nested static class

Commit:
Update History.java

Commit:
Update UUIDFetcher.java

lemonsked commented:
Converted, @JeromSar

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/UUIDFetcher.java:
```diff
@@ -0,0 +1,68 @@
+        return null;
+    }
+
+    static class FetchedUuid
```
Needs to be either private or public

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/util/History.java:
```diff
@@ -0,0 +1,104 @@
+        }
+    }
+
+    static class FName implements Comparable<FName>
```
Needs to be either private or public.

JeromSar reviewed (changes requested):
Just small changes now, and then we can test, and if all is well, merge.

Commit:
Update History.java

Commit:
Update UUIDFetcher.java

lemonsked commented:
Changes added @JeromSar

Wild1145 commented:
Currently building on TravisCI https://travis-ci.org/Wild1145/TotalFreedomMod/builds/237411850?utm_source=github_status&utm_medium=notification

JeromSar commented:
Just testing now.

Wild1145 reviewed (approved):
Build Passing TravisCI Tests

StevenLawson committed:
Fixed logviewer deactivate on setActive(true).

Wild1145 reviewed (dismissed):
There appear to be merge conflicts with the base branch now

Commit:
Merge branch 'server' into tfm5.1-mc1.10

lemonsked commented:
Resolved conflicts.

JeromSar commented:
Ouch, this will not go over well. The main development branch is currently `tfm5.1-mc1.12`. Syncing with `server` will cause problems.

Wild1145 commented:
@OxLemonxO Might want to revert the previous commit?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TotalFreedom/TotalFreedomMod?pullRequest=1847) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/TotalFreedom/TotalFreedomMod?pullRequest=1847) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: OxLemonxO<br/>:white_check_mark: marcocorriero<br/>:x: StevenLawson<br/><hr/>**marcocorriero** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TotalFreedom/TotalFreedomMod?pullRequest=1847) it.</sub>

lemonsked committed:
evert
t "Merge branch 'server' into tfm5.1-mc1.10"
Revert

This reverts commit a32d6db251f42322435c63844b3390718a349577, reversing
changes made to 293a876aeafea1e1ac11d37c9258998dd4d23e02.

lemonsked commented:
Reverted.

Commit:
Merge branch 'tfm5.1-mc1.12' into tfm5.1-mc1.10

ghost commented:
Tested, is broken.
@OxLemonxO 
@Wild1145

lemonsked commented:
@marcocorriero in which way?

Telesphoreo commented:
Fails for me also.
```[01:51:04 WARN]: [Fuse] Plugin Fuse v1.5 generated an exception while executing task 6
java.lang.NullPointerException: null
        at com.fusenetworks.fuse.util.UUIDFetcher.fetch(UUIDFetcher.java:47) ~[?:?]
        at com.fusenetworks.fuse.util.History$1.run(History.java:31) ~[?:?]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:71) ~[spigot-1.12.2.jar:git-Spigot-7754231-94b0980]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52) [spigot-1.12.2.jar:git-Spigot-7754231-94b0980]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]```

Thats what it is when implemented into Fuse

Wild1145 commented:
TravisCI doesn't appear to be kicking in for some reason... Apparently also needs the CLA?

lemonsked commented:
I've done CLA - no idea why it needs it again.

Commit:
Update UUIDFetcher.java

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TotalFreedom/TotalFreedomMod?pullRequest=1847) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/TotalFreedom/TotalFreedomMod?pullRequest=1847) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: OxLemonxO<br/>:white_check_mark: StevenLawson<br/>:x: marcocorriero<br/><hr/>**marcocorriero** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TotalFreedom/TotalFreedomMod?pullRequest=1847) it.</sub>

lemonsked commented:
Errors fixed.

Wild1145 commented:
Because @StevenLawson has a commit in the PR he will need to sign the CLA before the check passes.

Telesphoreo commented:
Still doesn't seem to work for me. It's now saying that it can't find the URL or something
```[01:12:25 ERROR]: null
java.io.FileNotFoundException: https://api.mojang.com/users/profiles/minecraft/
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1872) ~[?:1.8.0_131]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474) ~[?:1.8.0_131]
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254) ~[?:1.8.0_131]
        at com.fusenetworks.fuse.util.UUIDFetcher.fetch(UUIDFetcher.java:38) [Fuse.jar:?]
        at com.fusenetworks.fuse.util.History$1.run(History.java:31) [Fuse.jar:?]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:71) [spigot-1.12.2.jar:git-Spigot-7754231-94b0980]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52) [spigot-1.12.2.jar:git-Spigot-7754231-94b0980]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]

Commit:
This worked using Postman.... It shouldn't be erroring.

https://lemon.doesnt-have-a.life/pmqqg04.png

Telesphoreo commented:
Still gives this error
```[00:28:08 INFO]: [PLAYER_COMMAND] Telesphoreo (Telesphoreo): /namehistory Telesphoreo
[00:28:08 ERROR]: null
java.io.FileNotFoundException: https://api.mojang.com/users/profiles/minecraft/
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1872) ~[?:1.8.0_131]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474) ~[?:1.8.0_131]
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254) ~[?:1.8.0_131]
        at com.fusenetworks.fuse.util.UUIDFetcher.fetch(UUIDFetcher.java:38) [Fuse.jar:?]
        at com.fusenetworks.fuse.util.History$1.run(History.java:31) [Fuse.jar:?]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:71) [spigot-1.12.2.jar:git-Spigot-7754231-94b0980]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52) [spigot-1.12.2.jar:git-Spigot-7754231-94b0980]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]

Commit:
fix npe

Wild1145 commented:
Thank you for your contribution. Sadly we have decided to close this issue as there has been no recent activity to this pull request. If you wish to continue working on this PR please re-open the issue and ensure that all requested changes are made.
